packages/shared/sierra-client/src/index.ts

```diff
@@ -77,57 +105,89 @@ export default class SierraClient {
       },
       validateStatus: status => status === 200
     }).then(response => {
-      return axios.create({
+      const instance = axios.create({
         baseURL: this.apiRoot,
         headers: {
           Authorization: 'Bearer ' + response.data.access_token
-        },
+        }
+      });
+      instance.interceptors.response.use(response => response, error => {
+        return {
+          message: '[' + error.response.status + ' ' + error.response.statusText + "] " + error.response.data,
+          status: SierraStatus.UnknownError
+        };
       });
+      return instance;
     });
   }
 
   private toPatronRecord(data: any): PatronRecord {
-    const nameVarField = this.extractVarField(data.varFields, 'n');
-    const patronName = this.getPatronName(nameVarField);
+    const patronName = this.getPatronName(data.varFields);
     return Object.assign(patronName, {
         recordNumber: data.id,
-        barcode: this.extractVarField(data.varFields, 'b'),
-        emailAddress: this.extractVarField(data.varFields, 'z')
+        barcode: this.extractVarFieldContent(data.varFields, 'b'),
+        emailAddress: this.extractVarFieldContent(data.varFields, 'z')
       }
     );
   }
 
-  private extractVarField(varFields: VarField[], fieldTag: string) {
+  private extractVarFieldContent(varFields: VarField[], fieldTag: string) {
     const found = varFields.find(varField => varField.fieldTag === fieldTag);
     return found ? found.content : '';
   }
 
-  private getPatronName(varField: string): PatronName {
-    if (!varField.trim()) {
+  private getPatronName(varFields: VarField[]) {
```

Comment:
I think this should go out into it's own module so we can unit test deserialization of Sierra records.

Reply:
I've refactored the Patron models out into `patron.ts`.